module_prototyper/models/module_prototyper.py

```diff
@@ -0,0 +1,556 @@
+#
+##############################################################################
+import base64
+import logging
+import lxml.etree
+import os
+import re
+import textwrap
+
+from collections import namedtuple
+from datetime import date
+
+from jinja2 import Environment, FileSystemLoader
+
+from openerp import models, api, fields
+from openerp.tools.safe_eval import safe_eval
+
+from .default_description import get_default_description
+from . import licenses
+
+_logger = logging.getLogger(__name__)
+
+
+class ModulePrototyper(models.Model):
+    """Module Prototyper gathers different information from all over the
+    database to build a prototype of module.
+    We are calling it a prototype as it will most likely need to be reviewed
+    by a developer to fix glitch that would sneak it during the generation of
+    files but also to add not supported features.
+    """
+    _name = "module_prototyper"
+    _description = "Module Prototyper"
+
+    license = fields.Selection(
+        [
+            (licenses.GPL3, 'GPL Version 3'),
+            (licenses.GPL3_L, 'GPL-3 or later version'),
+            (licenses.LGPL3, 'LGPL-3'),
+            (licenses.LGPL3_L, 'LGPL-3 or later version'),
+            (licenses.AGPL3, 'Affero GPL-3'),
+            (licenses.OSI, 'Other OSI Approved Licence'),
+            ('Other proprietary', 'Other Proprietary')
+        ],
+        string='License',
+        default=licenses.AGPL3,
+    )
+    name = fields.Char(
+        'Technical Name', required=True,
+        help=('The technical name will be used to define the name of '
+              'the exported module, the name of the model.')
+    )
+    category_id = fields.Many2one('ir.module.category', 'Category')
+    human_name = fields.Char(
+        'Module Name', required=True,
+        help=('The Module Name will be used as the displayed name of the '
+              'exported module.')
+    )
+    summary = fields.Char('Summary', required=True,
+                          help=('Enter a summary of your module'))
+    description = fields.Text(
+        'Description',
+        required=True,
+        help=('Enter the description of your module, what it does, how to '
+              'install, configure and use it, the roadmap or known issues. '
+              'The description will be exported in README.rst'),
+        default=get_default_description
+    )
+    author = fields.Char('Author', required=True, help=('Enter your name'))
+    maintainer = fields.Char(
+        'Maintainer',
+        help=('Enter the name of the person or organization who will '
+              'maintain this module')
+    )
+    website = fields.Char('Website', help=('Enter the URL of your website'))
+    icon_image = fields.Binary(
+        'Icon',
+        help=('The icon set up here will be used as the icon '
+              'for the exported module also')
+    )
+    version = fields.Char(
+        'Version',
+        size=3,
```

Comment:
This idea is to have two digits only: major.minor. 

If migration is needed, change major, otherwise increment minor.